### ISSUE TITLE
Ability to paste rich text into the editor (and have it auto-convert to an element if no element is selected)

### ISSUE THREAD

**Issue by jauyong** [GoogleForCreators/web-stories-wp#760]
## Feature Description

As an editor I want to be able to paste rich text into the editor.

[Feature Brief
](https://docs.google.com/document/d/1_3-_v4zH6bW9hOplOO8wYxj8l4C0soMgEbD7XLiXrrg/edit#heading=h.y61q0s903u48)

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance Criteria
1. When not focused on an input, a new element should be created (with stripped away styling except for bold, etc.) (partly DONE in #1105, needs moving to global level)
2. We should strip away styling with the exception of bold, italic, underline and strikethrough (DONE in #1105)
2. When focused on text inputs, fallback to browser native cut/copy/paste
3. NOTE: strikethrough can come later, as it's not in the editor UI yet
4. See comments for required investigation for character sets

As of Apr 21:
![paste styling](https://user-images.githubusercontent.com/5874851/79874903-e9526880-83b6-11ea-89b1-fc6aace69b17.gif)

## QA Instructions

Please see notes in PR #5112.

**Comment by jauyong** [comment]
@pbakaus 
1. Should paste also include any rich styling (bold, hyperlink, etc...)?
2. What about character sets and special characters? (UTF8MB4)

**Comment by pbakaus** [comment]
@jauyong great questions:

1. We should strip away styling with the exception of bold, italic, underline and strikethrough ( strikethrough can come later, as it's not in the editor UI yet).

2. Actually not sure – when I copy text from a, say, ASCII encoding and paste it in the browser, does the browser not automatically convert/normalize to UTF-8? Might need some exploration.

**Comment by miina** [comment]
@jauyong To confirm -- based on the description, the issue seems to be about handling inputs only, if I understand correctly then it's actually the opposite:
When _not_ focused on an input, a new element should be created (with stripped away styling except for bold, etc.), that being the main goal/change of the issue. And when focused on inputs, the native browsers copy/paste should continue working as-is.

Let me know if I misunderstood.

**Comment by jauyong** [comment]
Hey @miina yea what you said is correct. I added your note as AC1.

**Comment by pbakaus** [comment]
@samitron7 I think this might warrant some Figma love for the final feature..your flow should describe what happens when I paste rich text from Google Docs, or a block of HTML with an image in it etc.

**Comment by miina** [comment]
Moving this back to todo since the basic pasting got partly implemented via #1105, however, there are still parts that need implementation, such as:
- Moving pasting to global level (related: #519)
- Improving pasting based on the results from UX.

**Comment by jauyong** [comment]
We are pushing this from sprint 27 / 1.0 Beta while we wait for the other text fixes that Morten is working on

**Comment by jauyong** [comment]
@samitron7 UX Needed:

> describe what happens when I paste rich text from Google Docs, or a block of HTML with an image in it etc.

**Comment by barklund** [comment]
This is not trivial. If it's not critical, I suggest icebox'ing this.

**Comment by pbakaus** [comment]
this is critical, so I vote to not icebox. Everyone working with the editor just a little will experience this, and be frustrated. Let's split this ticket though, the PO ask right now: ability to paste into a text box and preserve the styles that were in the text box before.

**Comment by o-fernandez** [comment]
The P0 bug @pbakaus refers to is handled by #1605 
This one, we can keep as P1 and do a proper analysis on the solution and effort and prioritize accordingly. Not crucial to fix in this sprint/release.

**Comment by csossi** [comment]
Verified in QA

Works as described in PR#5112 with some formatting NOT pasted

**Comment by samitron7** [comment]
Is there a reason why the rich text is pasted in centered aligned and not left justified for the pure canvas case? All our text presets are left aligned. @barklund FYI

**Comment by o-fernandez** [comment]
@barklund looks good, but @samitron7 has a question that would be worth addressing. I also noticed that it's all center aligned on paste, unless I first add a text element and then paste while in edit mode. 

Original in docs
![image.png](https://images.zenhubusercontent.com/5e98486db524b82fc96aeaf6/f8ccdf3c-35a5-4be8-8c97-b8b653acf669)

Pasted in canvas - first by just cmd+V on canvas, second by adding a paragraph text element and cmd+v while in text edit mode. 
![image.png](https://images.zenhubusercontent.com/5e98486db524b82fc96aeaf6/2d3ff4ba-f124-4375-802b-7290cf955351)

**Comment by o-fernandez** [comment]
Checked in UAT, still pastes as center aligned even if original content is not. Is this expected? @barklund @samitron7

**Comment by swissspidy** [comment]
Sounds unexpected to me. Would need a new ticket to resolve.

**Comment by o-fernandez** [comment]
@bmattb @barklund mind following up and ensuring there's a ticket/something to address this?

**Comment by barklund** [comment]
@o-fernandez , @samitron7: The default text set added by pressing the quick "+" next to the T in the library tab adds a new text field with center alignment. This same preset is used for pasted text so the alignment (and font size, family, etc) is taken from there.

**Comment by ivayloklissarov** [comment]
Tested and rich text works fine now. Closing the ticket. @tmagalhaes-huge
